dts/bindings/can/zephyr,native-linux-can.yaml

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022 Martin Jäger <martin@libre.solar>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Zephyr CAN driver using Linux SocketCAN
+
+compatible: "zephyr,native-linux-can"
+
+include: can-controller.yaml
+
+properties:
+    host-interface:
+      type: string
+      required: true
+      description: Linux host interface name (e.g. vcan0)
```

Comment:
I'd prefer not to make this binding in the devicetree as it may differ from host to host (not everybody would want to use a virtual CAN interface here). I'd suggest using the `native_add_command_line_opts()` function for adding a command-line argument to the native posix binary for setting the CAN device name instead.

Reply:
`vcan0` is only meant as an example for the most simple test case (without requiring any extra CAN hardware). For our use case we want to be able to have different CAN buses to implement a gateway (like CAN1 and CAN2 in an actual MCU) and not only a single one, so I'm not sure how we should tell the Zephyr application which host interface is connected to which internal device instance other than via devicetree.

Reply:
I know it's mentioned as an example in the binding, but it's also hardcoded in `boards/posix/native_posix/native_posix.dts`.

Reply:
Regarding your multi-interface use-case - couldn't the driver register a command-line argument for each instantiated interface?

Reply:
It could default to `vcan0` if needed.

Reply:
Would you be fine with an explanatory comment in the `.dts` file? I think it makes sense to provide the `vcan0` as an example because it can be tested on any Linux system.

Reply:
Oops, didn't see the comment regarding the multi-interface use case. Will have a look if that's possible.

Reply:
Had some more thoughts about the command-line approach and I still prefer to specify the Linux host CAN interface only via devicetree (overlays) to have a single source of truth. What happens if you compiled for two instances, but only pass one via the command line? Essentially you have to align the number of command line arguments to the number of actually defined instances in the devicetree, which makes the whole thing complicated and not more flexible.

If you want to change the interfaces without re-compiling the Zephyr binary, you can create custom aliases like `zephyrcan0` on the host which are used in the devicetree e.g. with `sudo ip link set vcan0 alias zephyrcan0`. That would be the equivalent of connecting the CAN wire from one pin to the other in real hardware.